Packs/Workday/Integrations/WorkdayIAMEventsGenerator/WorkdayIAMEventsGenerator.py

```diff
@@ -970,81 +827,158 @@
+                "Preferred_Name_-_Last_Name": "McBadbat",
+                "Cost_Center_Code": "651116",
+                "Location": "Office - USA - CA - Headquarters"
+            }
+        ]
+    }
+    new_report = report_entry['Report_Entry'][0]
+    for report in integration_context['Report_Entry']:
+        email_address = report.get('Email_Address')
+        if user_email == email_address:
+            return_error(f'User "{user_email}" already exist. Please try another user email.')
+
+    integration_context['Report_Entry'].append(new_report)
+    set_integration_context(integration_context)
 
 
 def get_terminate_report(user_email):
-    if not user_email:
-        set_integration_context(TERMINATE_USER_REPORT)
-    else:
-        integration_context = get_integration_context()
-        reports = TERMINATE_USER_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
-                users.append(report)
-                integration_context['Report_Entry'] = users
-                set_integration_context(integration_context)
+    integration_context = get_integration_context()
+    reports = TERMINATE_USER_REPORT['Report_Entry']
+    for report in reports:
+        email = report.get('Email_Address')
+        if email == user_email:
+            users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
+            users.append(report)
+            integration_context['Report_Entry'] = users
+            set_integration_context(integration_context)
+            return
+        else:
+            return_error(f'The user "{user_email}" does not exist. Please choose one of the following: '
+                         f'rrahardjo@test.com, sarnold@test.com, tfairy@test.com, rbuxaplenty@test.com, '
+                         f'cocarmichael@test.com, ccarmichael@test.com')
 
 
 def get_update_report(user_email):
-    if not user_email:
-        set_integration_context(UPDATE_REPORT)
-    else:
-        integration_context = get_integration_context()
-        reports = UPDATE_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
-                users.append(report)
-                integration_context['Report_Entry'] = users
-                set_integration_context(integration_context)
+    integration_context = get_integration_context()
+    reports = UPDATE_REPORT['Report_Entry']
+    for report in reports:
+        email = report.get('Email_Address')
+        if email == user_email:
+            users = [i for i in integration_context['Report_Entry'] if not (i['Email_Address'] == email)]
```

Comment:
the logic here is not very clear to me. When we want to generate update report, we should get an existing report and update some of it's fields.
I would recommend to add to this command few args like `title` and `city`. Also add `field_name` and `field_value` for update of any field